modules/aws/vault/vault-iam.tf

```diff
@@ -0,0 +1,58 @@
+resource "aws_iam_role" "vault" {
+  name = "${var.cluster_name}-vault"
+  path = "/"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "${var.iam_region}"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "vault-kms-unseal" {
+  statement {
+    sid       = "VaultKMSUnseal"
+    effect    = "Allow"
+    resources = ["arn:${var.arn_region}:iam::${var.aws_account}:user/${var.cluster_name}-vault-unseal-key"]
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:DescribeKey"
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "vault-ksm-unseal" {
+  name = "${var.cluster_name}-vault-kms-unseal"
+  role = "${aws_iam_role.vault.id}"
+  policy = "${data.aws_iam_policy_document.vault-kms-unseal.json}"
+}
+
+resource "aws_iam_instance_profile" "vault" {
+  name = "${var.cluster_name}-vault"
+  role = "${aws_iam_role.vault.name}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  # Sleep a little to wait the IAM profile to be ready -
+  provisioner "local-exec" {
+    command = "sleep 10"
```

Comment:
Look very hackish. Which is the reason for that? The reference is pretty explicit in the `aws_instance.vault` in the line

```hcl
  iam_instance_profile = "${aws_iam_instance_profile.vault.id}"
```

so Terraform will care about that.

Reply:
copy past from worker